sbomtool/cmd/sbomtool/main.go

```diff
@@ -0,0 +1,184 @@
+func printHelp() {
+	fmt.Println("sbomtool - Software Bill of Materials (SBOM) generation tool")
+	fmt.Println("\nUsage:")
+	fmt.Println("  sbomtool [global options] command [command options]")
+	fmt.Println("\nGlobal Options:")
+	fmt.Println("  --help               Show this help message")
+	fmt.Println("  --log-level string   Log level (debug, info, warn, error) (default \"info\")")
+	fmt.Println("\nCommands:")
+	fmt.Println("  generate    Generate SBOM files for a directory")
+	fmt.Println("  merge       Merge multiple SBOM files (not yet implemented)")
+	fmt.Println("  help        Show this help message")
+	fmt.Println("\nGenerate Options:")
+	fmt.Println("  --name string        Name of the target package")
+	fmt.Println("  --build-dir string   Target directory of the package")
+	fmt.Println("  --out-dir string     Output directory for the SBOM")
+	fmt.Println("  --spdx               Generate an SPDX SBOM")
+	fmt.Println("  --cyclonedx          Generate a CycloneDX SBOM")
+	fmt.Println("\nMerge Options:")
+	fmt.Println("  --level int          Merge level (default 0)")
+	fmt.Println("\nExamples:")
+	fmt.Println("  sbomtool --log-level debug generate --name myapp --build-dir ./build --out-dir ./sbom --spdx")
+	fmt.Println("  sbomtool generate --name myapp --build-dir ./build --out-dir ./sbom --cyclonedx")
+}
+
+// findSubcommandPos finds the position of the first non-flag argument (the subcommand)
+func findSubcommandPos() int {
+	for i := 1; i < len(os.Args); i++ {
+		if !strings.HasPrefix(os.Args[i], "-") {
+			return i
+		}
+	}
+	return len(os.Args)
+}
+
+// run executes the main application logic and returns an exit code.
+// This separation allows for better testability by avoiding direct os.Exit calls.
+func run() int {
+	// Define global flags
+	globalFlags := flag.NewFlagSet("sbomtool", flag.ExitOnError)
+	helpFlag := globalFlags.Bool("help", false, "Show help message")
+	logLevelFlag := globalFlags.String("log-level", "info", "Log level (debug, info, warn, error)")
+
+	// Find the position of the first non-flag argument (the subcommand)
+	subcommandPos := findSubcommandPos()
+
+	// If there are no arguments or only flags, parse them
+	if subcommandPos == len(os.Args) {
+		globalFlags.Parse(os.Args[1:])
+		// Show help if no subcommand is provided or help flag is set
+		if *helpFlag || subcommandPos == len(os.Args) {
+			printHelp()
+			return 0
+		}
+	} else {
+		// Parse only the global flags before the subcommand
+		globalFlags.Parse(os.Args[1:subcommandPos])
+	}
+
+	// Configure logging based on global flag
+	logging.Configure(logging.LogLevel(*logLevelFlag))
+
+	// Show help if help flag is set
+	if *helpFlag {
+		printHelp()
+		return 0
+	}
+
+	// If no subcommand is provided or "help" is the subcommand
+	if subcommandPos == len(os.Args) || os.Args[subcommandPos] == "help" {
+		printHelp()
+		return 0
+	}
+
+	// Get the subcommand
+	subcommand := os.Args[subcommandPos]
+
+	switch subcommand {
```

Comment:
I know `cobra` is a widely used CLI tool in Go, but I'll let the Go experts advise whether adding that to this CLI would be unnecessary.